Add overview content to token challenge section (#479)
ietf-wg-privacypass/base-drafts: draft-ietf-privacypass-auth-scheme.md

## Token Challenge {#challenge}

Origins send a token challenge to clients in an "WWW-Authenticate" header field

> with the "PrivateToken" scheme. This field is a series parameters, each a key-value pair. There are two mandatory parameters:
> one containing a token challenge and another containing the token-key used for

Reviewer: ```suggestion
with the "PrivateToken" scheme. This authentication scheme has two mandatory parameters:
one containing a token challenge and another containing the token-key used for
```

Reviewer: Since this is just part of how auth schemes work...

> producing (and verifying) a corresponding token. The challenge parameter can be
> repeated in the header field.

Reviewer: The challenge parameter isn't repeated, the entire authentication challenge with all parameters is repeated. Again, this is just how auth schemes work.

```suggestion
producing (and verifying) a corresponding token.
```

Author: Right, right.

## Token Challenge {#challenge}

> Origins send a token challenge to clients in an "WWW-Authenticate" header field

Reviewer: I wonder if we should mention "(or "Proxy-Authenticate")

Author: That seems like an increase in scope, so if we do we'd probably want to run it past the list.

Reviewer: Yeah we don't necessarily need to mention that — it was just alluded to in one of martin's comments

with the "PrivateToken" scheme. This field is a series of key-value pairs, where

> values are base64url-encoded binary data. Each field has two mandatory parameters,

Reviewer: Not all parameters to the auth scheme will necessarily be base64... I wonder if we shouldn't go into the key-value bit, since that's normal to an auth scheme.

> with the "PrivateToken" scheme. This field is a series of key-value pairs, where
> values are base64url-encoded binary data. Each field has two mandatory parameters,

Author: ```suggestion
with the "PrivateToken" scheme. This field is a series parameters, each a key-value pair. There are two mandatory parameters:
```
